src/client/store/ui/geo/slice.ts

```diff
@@ -33,6 +33,20 @@ const initialState: GeoState = {
   },
 }
 
+const getSectionState = (state: Draft<GeoState>, sectionKey: LayerSectionKey): LayersSectionState => {
+  // Default the states to an empty object if they don't exist yet
+  state.sections[sectionKey] ??= {} as LayersSectionState
+  return state.sections[sectionKey]
+}
+
+const getLayerState = (state: Draft<GeoState>, sectionKey: LayerSectionKey, layerKey: LayerKey): LayerState => {
+  // Default the states to an empty object if they don't exist yet
+  state.sections[sectionKey] ??= {} as LayersSectionState
```

Comment:
You can call getSectionState here

Reply:
I kept it like this for readability, but now that you mention, if we are ever to change the way we initialize sections, it's better to call the getSectionState. Thanks, this is now updated.